### ISSUE TITLE
Pickling typing.List currently fails

### ISSUE THREAD

**Issue by robertnishihara** [amplab/ray#293]
On master, start `python scripts/shell.py`. Then do

```python
import typing

@ray.remote([typing.List], [int])
def f(xs):
  return sum(xs)
```
The definition currently fails with
```python
Traceback (most recent call last):
  File "/Users/rkn/Workspace/ray/scripts/default_worker.py", line 28, in <module>
    ray.worker.main_loop()

  File "/Users/rkn/Workspace/ray/lib/python/ray/worker.py", line 695, in main_loop
    (function, arg_types, return_types) = pickling.loads(command_args)
  File "/Users/rkn/anaconda/lib/python2.7/pickle.py", line 1388, in loads
    return Unpickler(file).load()
  File "/Users/rkn/anaconda/lib/python2.7/pickle.py", line 864, in load
    dispatch[key](self)
  File "/Users/rkn/anaconda/lib/python2.7/pickle.py", line 1139, in load_reduce
    value = func(*args)
  File "/Users/rkn/Workspace/ray/lib/python/ray/pickling.py", line 42, in _create_type
    return eval(type_repr.replace("~", ""), None, (lambda d: d.setdefault("typing", typing) and None or d)(dict(typing.__dict__)))
  File "<string>", line 1
    typing.List<T>
                 ^
SyntaxError: unexpected EOF while parsing
```
Note that this error can be "fixed" by commenting out the last line of `pickling.py`
```python
#dispatch[typing.GenericMeta] = save_type
```
The purpose of that code was to address cloudpipe/cloudpickle#63, right?

**Comment by mehrdadn** [comment]
Is this a regression? It works fine on my machine (re-tested it right now).
Things like `typing.List[int]` appear with brackets on my machine, not carets though... are carets a Mac thing?

**Comment by robertnishihara** [comment]
I just tried it on a clean install of Ubuntu (with the current master branch), and I get the same error. I also get the carets.

**Comment by robertnishihara** [comment]
Interestingly, I just tried replacing `List` with `List[int]` in the above example and that worked.

**Comment by mehrdadn** [comment]
Hm... I can't repro this on my Ubuntu...

	In [1]: import typing
	
	In [2]: @ray.remote([typing.List], [int])
	def f(xs): return sum(xs)
	   ...:
	
	In [3]: ray.get(f([1, 2, 3]))
	1 task succeeded
	1 task succeeded
	Out[3]: 6

**Comment by mehrdadn** [comment]
Oh, I just realized it's because you commented out that line! Never mind...

**Comment by mehrdadn** [comment]
Addressed in #315.
